### PR TITLE
Enhance edit Command: Batch Edit for All Fields (Except Name), Code Quality & Error Message Improvements

### PR DESCRIPTION
**Pull request by soakedumplings**
<head></head>
<h3><strong>Summary</strong></h3>
<p>
This PR extends the functionality of the <code>edit</code> command to support <strong>batch editing</strong> for all editable fields <strong>except <code>Name</code></strong>, improves <strong>code quality and adherence to SLAP (Single Level of Abstraction Principle)</strong> for <code>EditCommand</code> and <code>EditCommandParser</code>, and refines <strong>error messages</strong> to enhance user clarity.
</p>

<hr>

<h3><strong>Key Enhancements</strong></h3>

<h4><strong>Functional Improvements</strong></h4>
<ul>
  <li>
    <p>
      Implemented <strong>batch editing</strong> capability for all fields (<code>Phone</code>, <code>Email</code>, <code>Address</code>, <code>Status</code>, <code>Remark</code>, <code>Tag</code>) except <code>Name</code>.
    </p>
    <ul>
      <li>
        <p>Example:</p>
        <pre><code>edit 1,2,3 e/newemail@example.com
</code></pre>
        <p>updates the email for companies 1, 2, and 3.</p>
      </li>
    </ul>
  </li>
  <li>
    <p>Ensured that <strong>field-specific edit logic</strong> applies consistently across both single and batch edits.</p>
  </li>
  <li>
    <p>Added <strong>robust validation and graceful error handling</strong> when conflicting or invalid inputs are detected.</p>
  </li>
</ul>

<hr>

<h4><strong>Code Quality &amp; SLAP</strong></h4>
<ul>
  <li>
    <p>Refactored <code>EditCommand</code> and <code>EditCommandParser</code>:</p>
    <ul>
      <li><p>Extracted helper methods for clearer separation of concerns.</p></li>
      <li><p>Simplified conditional logic for better readability and maintainability.</p></li>
      <li><p>Enhanced naming conventions to improve semantic clarity.</p></li>
    </ul>
  </li>
  <li>
    <p>Removed redundant code and improved overall structure to comply with <strong>SLAP and SE-EDU coding standards</strong>.</p>
  </li>
</ul>

<hr>

<h4><strong>Error Message Improvements</strong></h4>
<ul>
  <li>
    <p>Revised error messages to be more <strong>specific, user-friendly, and consistent</strong>:</p>
    <ul>
      <li><p>Differentiated messages for invalid index ranges, duplicate field edits, and missing prefixes.</p></li>
      <li><p>Added contextual hints (e.g., “Ensure at least one editable field is provided”).</p></li>
      <li><p>Unified tone and formatting for all edit-related messages.</p></li>
    </ul>
  </li>
</ul>

<hr>

<h3><strong>Testing</strong></h3>
<ul>
  <li>
    <p>Added/updated <strong>JUnit tests</strong> to cover:</p>
    <ul>
      <li><p>Batch edit functionality across all fields.</p></li>
      <li><p>Invalid input handling and edge cases.</p></li>
      <li><p>Parser behavior for multi-index and multi-field inputs.</p></li>
    </ul>
  </li>
  <li><p>Verified test coverage meets project standards.</p></li>
</ul>

<hr>

<h3><strong>Example Usage</strong></h3>
<table>
  <tr>
    <th>Command</th>
    <th>Description</th>
  </tr>
  <tr>
    <td><code>edit 1,2 p/91234567</code></td>
    <td>Updates phone for companies 1 and 2</td>
  </tr>
  <tr>
    <td><code>edit 1,2,3 s/applied</code></td>
    <td>Updates status for companies 1–3</td>
  </tr>
  <tr>
    <td><code>edit 5 r/Updated remark</code></td>
    <td>Updates remark for company 5</td>
  </tr>
</table>

<hr>

<h3><strong>Rationale</strong></h3>
<p>
This enhancement improves <strong>usability</strong>, <strong>code maintainability</strong>, and <strong>error transparency</strong>, enabling users to efficiently manage multiple records while maintaining clear and robust system behavior.
</p>

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -52,39 +53,29 @@ public class EditCommand extends Command {
...
+            + "Example (batch edit - allowed for all fields except Name): " + COMMAND_WORD + " 1,2,4-8 "
+            + PREFIX_PHONE + "91234567 "
+            + PREFIX_EMAIL + "googlehr@gmail.com "
+            + PREFIX_ADDRESS + "70 Pasir Panjang Rd, #03-71 Mapletree Business City II, Singapore 117371 "
+            + PREFIX_STATUS + "applied "
+            + PREFIX_TAG + "FAANG";
 
-    public static final String MESSAGE_EDIT_COMPANY_SUCCESS = "Edited Company: %1$s";
-    public static final String MESSAGE_BATCH_EDIT_SUCCESS = "Edited %1$d companies successfully";
+    public static final String MESSAGE_EDIT_SUCCESS = "Edited %1$d companies successfully";
```
Comment (kzfsg) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: Why is the naming convention less specific now?

**Review by kzfsg — APPROVED** [review]
Mostly nits, can discuss on my comments as a group then reconvene & merge.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -119,22 +112,15 @@ public CommandResult execute(Model model) throws CommandException {
...
         Company editedCompany = createEditedCompany(companyToEdit, editCompanyDescriptor);
 
-        if (!companyToEdit.isSameCompany(editedCompany) && model.hasCompany(editedCompany)) {
-            throw new CommandException(MESSAGE_DUPLICATE_COMPANY);
-        }
-
         model.setCompany(companyToEdit, editedCompany);
         model.updateFilteredCompanyList(PREDICATE_SHOW_ALL_COMPANIES);
-        return new CommandResult(String.format(MESSAGE_EDIT_COMPANY_SUCCESS, Messages.format(editedCompany)));
+        return new CommandResult(String.format(MESSAGE_EDIT_SUCCESS, indices.size()));
```
Comment (jeffreylim24) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: I like that you're trying to keep consistency, but I think in the future it will be good to show which companies were edited too, but that's for another PR

**Review by jeffreylim24 — APPROVED** [review]
LGTM! Just a few nits
